frontend/app/services/lookup-service.server.ts

```diff
@@ -593,21 +586,46 @@ function createLookupService() {
     const url = `${INTEROP_API_BASE_URI}/lookups/marital-statuses`;
     const response = await fetch(url);
 
-    const maritalStatusSchemaList = z.array(maritalStatusSchema);
+    if (!response.ok) {
+      log.error('%j', {
+        message: 'Failed to fetch data',
+        status: response.status,
+        statusText: response.statusText,
+        url: url,
+        responseBody: await response.text(),
+      });
 
-    if (response.ok) {
-      return maritalStatusSchemaList.parse(await response.json());
+      throw new Error(`Failed to fetch data. Status: ${response.status}, Status Text: ${response.statusText}`);
     }
 
-    log.error('%j', {
-      message: 'Failed to fetch data',
-      status: response.status,
-      statusText: response.statusText,
-      url: url,
-      responseBody: await response.text(),
+    const maritalStatusSchema = z.object({
+      value: z.array(
+        z.object({
+          OptionSet: z.object({
+            Options: z.array(
+              z.object({
+                Value: z.number(),
+                Label: z.object({
+                  LocalizedLabels: z.array(
+                    z.object({
+                      Label: z.string(),
+                    }),
+                  ),
+                }),
+              }),
+            ),
+          }),
+        }),
+      ),
     });
 
-    throw new Error(`Failed to fetch data. Status: ${response.status}, Status Text: ${response.statusText}`);
+    const parsedMaritalStatusData = maritalStatusSchema.parse(await response.json());
+    return parsedMaritalStatusData.value[0].OptionSet.Options.map((o) => ({
+      id: o.Value,
+      code: o.Value,
```

Comment:
**Nitpick/Suggestion:** Would it make sense for the service to return strings for `id`/`code` so we don't have to use `toString()` everywhere when referencing these fields?